src/c/core/session/stream/output_reliable_stream_internal.h

```diff
@@ -28,11 +28,9 @@ extern "C"
 
 #define HEARTBEAT_PAYLOAD_SIZE 4
 
-struct ucdrBuffer;
-
-void uxr_init_output_reliable_stream(uxrOutputReliableStream* stream, uint8_t* buffer, size_t size, uint16_t history, uint8_t header_offset);
+void uxr_init_output_reliable_stream(uxrOutputReliableStream* stream, uint8_t* buffer, size_t size, uint16_t history, uint8_t header_offset, OnNewFragment on_new_fragment);
 void uxr_reset_output_reliable_stream(uxrOutputReliableStream* stream);
-bool uxr_prepare_reliable_buffer_to_write(uxrOutputReliableStream* stream, size_t size, struct ucdrBuffer* ub);
+bool uxr_prepare_reliable_buffer_to_write(uxrOutputReliableStream* stream, size_t size, size_t fragment_offset, struct ucdrBuffer* ub);
```

Comment:
Why is `fragment_offset` a paramenter. This function is only used in `session.c` with `SUBHEADER_SIZE` as `fragment_offset`.

Reply:
`fragment_offset` is used to take into account the offset produced by the bytes related with the metainformation about the fragment. For the current implementation, session uses SUBHEADER_SIZE bytes as offset for its implementation. Future or another implementation that uses fragmentation can be codified in another way.

Reply:
It sounds weird. You do think it is useful to keep this as a parameter for future implementations?

Reply:
I need the parameter because the size of the total data into the stream must be calculated without the size of the fragmentation. Is the less weird way I have found.

Reply:
Ook!